### ISSUE TITLE
Monitoring: Grafana not accessible in EKS cluster

### ISSUE THREAD

**Issue by lbogdan** [rancher/rancher#19740]
**TL;DR**: Is there some docs describing how service proxying works in rancher? I would like to debug this myself, but I have no idea how things are connected and where should I start. Also, I'm guessing it might have to do with the fact that the EKS nodes are completely firewalled from the outside world? (which is the only difference between the EKS cluster and a bare-metal cluster with no firewall, which works fine)

**Related**: rancher/rancher#16968 (Monitoring - Grafana links not showing up for EKS)

**What kind of request is this (question/bug/enhancement/feature request):**
Question / bug?

**Steps to reproduce (least amount of steps as possible):**
- create EKS cluster (using pulumi, maybe that's relevant)
- run rancher external to AWS
- import the EKS cluster into rancher
- enable monitoring
- try to access Grafana (using `https://rancher.url/k8s/clusters/c-abcde/api/v1/namespaces/cattle-prometheus/services/http:access-grafana:80/proxy/` URL)

**Result:**
```
Error: 'dial tcp a.b.c.d:80: connect: connection timed out'
Trying to reach: 'http://a.b.c.d:80/'
```
where `a.b.c.d` is the IP address of the `grafana-cluster-monitoring` pod.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): **v2.2.2**
- Installation option (single install/HA): **single**

I would appreciate if you could point me in the right direction to be able to debug this myself. Also, please let me know if you need additional information.

**Comment by vincent99** [comment]
The server container proxies `/k8s/clusters/c-abcde` to a control plane node in that cluster, and the rest (`/api/v1/namespaces/cattle-prometheus/services/http:access-grafana:80/proxy/`) is part of standard k8s to access a service.

**Comment by JimTim** [comment]
I had the same problem and this was my workaround:

access-grafana service:
```yaml
...
spec:
  clusterIP: ...
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: http
...
```

grafana-cluster-monitoring deployment:
```yaml
    spec:
      containers:
      - env:
        - name: GF_AUTH_BASIC_ENABLED
          value: "true"
        - name: GF_AUTH_ANONYMOUS_ENABLED
          value: "true"
        image: rancher/grafana-grafana:5.4.3
        imagePullPolicy: IfNotPresent
        name: grafana
        ports:
        - containerPort: 3000
          name: web
          protocol: TCP
```

You can see the error: In deployment the port is named with "web" and the service referenced http. So as workaround change the service port name to web and it will work. 

But this is not the final solution. Where can I find the helm charts for the system apps?

**Comment by lbogdan** [comment]
@JimTim the `grafana-cluster-monitoring` deployment also has a proxy container
```yaml
      - args:
        - nginx
        - -g
        - daemon off;
        - -c
        - /nginx/nginx.conf
        image: rancher/nginx:1.15.8-alpine
        imagePullPolicy: IfNotPresent
        name: grafana-proxy
        ports:
        - containerPort: 80
          name: http
          protocol: TCP
        [...]
        volumeMounts:
        - mountPath: /nginx/
          name: grafana-nginx
```
which is the one the service points to, so that's not the issue.

**Comment by lbogdan** [comment]
In my case, the issue was that the security group that deals with the traffic between the k8s API server (managed by EKS) and a node instance was setup by `pulumi` to allow only ports between 1025-65535, so when trying to proxy `access-grafana:80`, which points to `grafana-cluster-monitoring-pod:80`, that was denied. The quick solution was to extend the port range to 80-65535.

**Comment by wreed4** [comment]
That worked for me as well @lbogdan !  I'm creating my cluster/vpc with terraform using the aws vpc module.  I don't suppose anyone knows off the top of their heads how to automate this security group change so it doesn't need to be modified later?

**Comment by lbogdan** [comment]
For future reference, from https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html :

> **Note**
>
> To allow proxy functionality on privileged ports or to run the CNCF conformance tests yourself, you must edit the security groups for your control plane and the worker nodes. The security group on the worker nodes' side needs to allow inbound access for ports 0-65535 from the control plane, and the control plane side needs to allow outbound access to the worker nodes on ports 0-65535.

**Comment by loganhz** [comment]
Thanks @lbogdan 
Let's move the talk to https://github.com/rancher/rancher/issues/16968

Close in favor of https://github.com/rancher/rancher/issues/16968
